app/views/results/common/_context_menu.js.erb

```diff
@@ -1,113 +1,46 @@
-
-    Context-Menu button functionality:
-      * _annotations.js.erb (..\app\views\results\common\)
-
-     */
   (function() {
-    var released_to_students = <%= @result.released_to_students %>;
+    var released_to_students = <%= result.released_to_students %>;
     if (released_to_students) return;
-    var menu_items = {
-      new_annotation: {
-        title: 'New Annotation',
-        cmd: 'new_annotation',
-        action: make_new_annotation,
-        disabled: true
-      },
-      delete_annotation: {
-        title: 'Delete Annotation',
-        cmd: "delete_annotation",
-        action: function(event, ui) {
-          var clicked_element = $(ui.target);
-          var annot_id = '';
-          var sub_file_id = $('#select_file_id').val();
-          if (annotation_type === ANNOTATION_TYPES.CODE) {
-            $.each(clicked_element[0].attributes, function(index, attr) {
-              if (attr.nodeName.toLowerCase()
-                      .indexOf('data-annotationid') != -1) {
-                annot_id = attr.value;
-                // Continue iteration in case of multiple annotations
-              }
-            });
-          } else {
-            annot_id = clicked_element.attr('id')
-                                      .replace('annotation_holder_', '');
-          }
-
-          if (annot_id !== null && annot_id.length !== 0 &&
-              sub_file_id !== null && sub_file_id.length !== 0) {
-            var query_str_params = [
-              ['id=' + annot_id],
-              ['submission_file_id=' + sub_file_id],
-              ['result_id=<%= @result.id %>'],
-              ['assignment_id=<%= @result.submission.grouping.assignment.id %>']
-            ];
-            $.ajax({
-              url: '<%= annotations_path %>?' + query_str_params.join('&'),
-              method: 'POST',
-              data: { _method: 'delete' },
-              dataType: 'script'
-            });
 
+    var annot_path = '<%= annotations_path %>';
+    var result_id = '<%= result.id %>';
+    var assignment_id = '<%= result.submission.grouping.assignment.id %>';
+    var file_dl_path = '<%= download_assignment_submission_result_path(
+      result.submission.grouping.assignment.id,
+      result.submission.grouping.current_submission_used.id,
+      result.submission.grouping
+            .current_submission_used.get_latest_result.id) %>';
+    var common_annotations = [
+    <% annotation_categories.each do |annotation_category| %>
+      {
+        title: '<%= truncate(annotation_category.annotation_category_name, length: 40) %> <kbd>></kbd>',
+        cmd: 'annotation_category_<%= annotation_category.id %>',
+        action: function(event, ui){ return false; },
+        children: [
+        <% if annotation_category.annotation_texts.empty? %>
+          {
+            title: '<%= t('marker.annotation.no_annotation_in_this_category') %>',
+            action: function(event, ui){ return false; }
```

Comment:
Since these items are going to be submenus/children of `Common Annotations`, having `disabled: true` should not be necessary here since its parent will be disabled unless the conditions correct for the children as well as the parent.